chain/x/zoracle/handler.go

```diff
@@ -151,7 +151,10 @@ func handleMsgRequestData(ctx sdk.Context, keeper Keeper, msg MsgRequestData) sd
 	if err != nil {
 		return err.Result()
 	}
-	_, _, errOwasm := owasm.Execute(&env, script.Code, "prepare", msg.Calldata, 100000)
+
+	ctx.GasMeter().ConsumeGas(msg.PrepareGas, "PrepareRequest")
```

Comment:
Let's make a standard to consume gas. In `handleEndBlock` we consume gas after call execution. @sorawit please help us to decide what it will be.

Reply:
We should consume gas before doing work

Reply:
Make sense!